core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1696,12 +1696,12 @@ public function generateRecordLabel(array $row, ?string $table = null, bool $pro
 			{
 				$row[$v] = Encryption::decrypt(StringUtil::deserialize($row[$v]));
 			}
-
+			$objDcaExtractor = DcaExtractor::getInstance($table);
+			$strRelationField = ($objDcaExtractor->getRelations()[$v]['field'] ?? 'id');
 			if (strpos($v, ':') !== false)
 			{
 				list($strKey, $strTable) = explode(':', $v, 2);
 				list($strTable, $strField) = explode('.', $strTable, 2);
-				$strRelationField = Database::quoteIdentifier($GLOBALS['TL_DCA'][$table]['fields'][$v]['relation']['field'] ?? 'id');
 				$objRef = Database::getInstance()
 					->prepare("SELECT " . Database::quoteIdentifier($strField) . " FROM " . $strTable . " WHERE " .$strRelationField ."=?")
```

Comment:
This does not work I think for label configs like `user_id:tl_user.name` as `$objDcaExtractor->getRelations()[$v]` would always be null then.

Reply:
Changed to
`$strRelationField = $objDcaExtractor->getRelations()[$strKey]['field'] ?? 'id';`

Also moved the quoteIdentifier call to the SQL line to match the coding style of all the rest of the SQL statements.